packages/cli/src/lib/merge/index.js

```diff
@@ -0,0 +1,236 @@
+  traverse(ast, {
+    Program(path) {
+      programPath = path
+      return
+    },
+  })
+  if (programPath === undefined) {
+    throw new Error('Unable to find Program node in AST')
+  }
+  return programPath
+}
+
+// See https://github.com/babel/babel/issues/14480
+function skipChildren(path) {
+  for (const key of VISITOR_KEYS[path.type]) {
+    path.skipKey(key)
+  }
+}
+
+/**
+ * We can make merge strategies more terse and intuitive if we pass a Babel Node, rather than a
+ * NodePath, to the reducer. This would allow us to write:
+ *
+ * ArrayExpression: (lhs, rhs) => { lhs.elements.push(...rhs.elements) }
+ * instead of
+ * ArrayExpression: (lhs, rhs) => { lhs.node.elements.push(...rhs.node.elements) }
+ *
+ * It may seem like a small difference, but the code is much more intuitive if you don't have to
+ * think about Babel nodes vs paths when writing reducers.
+ *
+ * We could just pass the node directly to the reducer, but there are reasonable (though rare) cases
+ * where you do want access to the NodePath. To solve this, we create a proxy object that appears as
+ * a Babel Node with an additional `path` property that points back to the NodePath.
+ */
+function makeProxy(path) {
+  return new Proxy(path, {
+    get(target, property) {
+      if (property === 'path') {
+        return target
+      } else {
+        return target.node[property]
+      }
+    },
+    set(target, property, value) {
+      if (property === 'path') {
+        throw new Error("You can't set a path on a proxy!")
+      } else {
+        target.node[property] = value
+        return true
+      }
+    },
+    has(target, property) {
+      return property in target.node
+    },
+  })
+}
+
+function expressionUses(exp, ...ids) {
+  let result = false
+  exp.traverse({
+    Identifier(path) {
+      if (
+        !path.parentPath.isNodeType('VariableDeclarator') &&
+        ids.includes(path.node.name)
+      ) {
+        result = true
+        return
+      }
+    },
+  })
+  return result
+}
+
+// Insert the given expression before the first usage of its name in 'path', or at the end of the
+// program body if no such usage exists.
+function insertBeforeFirstUsage(expression, program) {
+  const body = program.get('body')
+  const pos = body.findIndex((exp) =>
+    expressionUses(exp, ...Object.keys(expression.getBindingIdentifiers()))
+  )
+  return pos !== -1
+    ? body[pos].insertBefore(expression.node)
+    : program.pushContainer('body', expression.node)
+}
+
+function insertAfterLastImport(expression, program) {
+  const body = program.get('body')
+  return body[
+    body.findLastIndex((bodyExpr) => bodyExpr.isNodeType('ImportDeclaration'))
+  ].insertAfter(expression.node)
+}
+
+function prune(path) {
+  switch (path.parentPath.type) {
+    // If pruning 'path' would yield an ill-formed parent (e.g, '{foo:}' or 'const x;'), prune it.
+    case 'ObjectProperty':
+    case 'VariableDeclarator':
+      return path.parentPath.remove()
+    default:
```

Comment:
I don't think I've ever seen a switch statement before where `default` wasn't last.

It might not be as DRY, but I'd prefer to move the `default` to the end and have `path.remove()` repeated in it.